### DISCUSSION TITLE
How does "NULL mapping to pointer to pointer" work?

### DISCUSSION THREAD

**Discussion by lzap** [jackc/pgx#1322] [General]
Hey,

I am converting our project from database/sql and sqlx to pgx/scany. We use sql.NullXXX for some types and I saw these in pgx tests too, so I assume they will work. But I also saw "NULL mapping to pointer to pointer" in the README. Tho I can't find anything relevant in the codebase

How does this work?

Is it any better than sql.NullXXX types? 

Does it work nicely with JSON marshaling?

Thanks!

**Comment by lzap** [comment]
The documentation is lacking, I was thinking maybe contributing few wiki pages with basics, do you like the idea or maybe you have some other plans (like `Mastering pgx` book in the works). :)

**Comment by jackc** [comment]
Pointer to pointer just the following:

```go
var s *string
conn.QueryRow(...).Scan(&s)
```

If a NULL is scanned then s will be set to nil. If a non-NULL is scanned then a string will be allocated and s will pointer to it.

**Reply by lzap** [reply]
Is it that easy? I would expect dragons since the Go standard library authors figured the beefy NullXY. Okay then :-)

**Reply by sai-sy** [reply]
How would this work across struct fields, for example converting this to handle null
```go
persons := []person{}
  count := 0
  for rows.Next() {
    person := person{}
    err := rows.Scan(
      &person.UUID,
```

because when you make this change:
```go
}
  count := 0
  for rows.Next() {
    var person *person
    err := rows.Scan(
      &person.UUID,
      &person.Created_at,
      &person.Givenname, 
```
gives a nil dereference warning

**Reply by jackc** [reply]
@sai-sy You can't take the address of a field of `person` when `person` is `nil`.
